tests/integration/monitoring/test-monitoring.go

```diff
@@ -152,13 +150,12 @@ func testTargetsAreHealthy() {
 			allTargetsAreHealthy := true
 			timeoutMessage = ""
 			for _, target := range activeTargets {
-				// Ignoring the targets with certain labels
-				if hasAnyLabel(target.Labels, labelsToBeIgnored) {
+				if shouldIgnoreTarget(target.Labels) {
 					continue
 				}
 				if target.Health != "up" {
 					allTargetsAreHealthy = false
-					timeoutMessage += fmt.Sprintf("Target with job=%s and instance=%s is not healthy\n", target.Labels["job"], target.Labels["instance"])
+					timeoutMessage += fmt.Sprintf("Target with job=%s and instance=%s is not healthy, with labels=%v\n", target.Labels["job"], target.Labels["instance"], target.Labels)
```

Comment:
A good candidate for a `strings.Builder`.

Reply:
I agree. There are other occurrences of this pattern that need to be handled holistically in the file. Hence I would keep it as is for the sake of consistency.